portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,11 +53,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Comment comment = new Comment(id, name, text, timestamp);
       comments.add(comment);
     }
+    return comments;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JsonObject res = new JsonObject();
+
+    try {
+      int maxComments = Integer.parseInt(request.getParameter("max-comments"));
+      List<Comment> comments = queryComments(maxComments);
+
+      Gson gson = new Gson();
+
+      JsonParser parser = new JsonParser();
+      JsonElement commentsJson = parser.parse(gson.toJson(comments));
+
+      res.add("comments", commentsJson);
+      res.addProperty("success", true);
+      } catch (Exception e) {
```

Comment:
You can catch the right class of the exception rather than rely on the string of the error. The error message should not be relied on for error handling.

For example your catch can be

catch(NumberFormatException e) {
 // Handle number format exception here.
}